rebasehelper/tests/test_utils.py

```diff
@@ -42,6 +44,32 @@
 from rebasehelper.utils import LookasideCacheHelper
 
 
+class TestGitHelper(object):
+
+    @pytest.yield_fixture
+    def temp_dir(self):
+        current_dir = os.getcwd()
+        try:
+            with TemporaryEnvironment() as temp:
+                temp_repo_path = temp.path()
+                os.chdir(temp_repo_path)
+                yield
+        finally:
+            os.chdir(current_dir)
+
+    @pytest.mark.usefixtures('temp_dir')
+    def test_get_user_and_email(self):
+        temp_repo_path = os.getcwd()
+        repo = git.Repo.init(temp_repo_path)
+        name = 'Foo Bar'
+        email = 'foo@bar.com'
+        repo.git.config('user.name', name, local=True)
+        repo.git.config('user.email', email, local=True)
```

Comment:
This only tests local-level configuration. It would be good to test also global-level and user-level, and also the case with include path. But I'm not sure how to do that, we certainly don't want tests to overwrite someone's configuration.

Reply:
You could for example create a set of config files (include one in the other etc.) and then export `GIT_CONFIG` environment variable so that those files are read by `GitHelper.get_user()` and `GitHelper.get_email()`.

Reply:
@nforro Using `GIT_CONFIG` is same with `-f config-file` option. It is not `--global`'s flow.
In case of `--global`, using `$XDG_CONFIG_HOME/git/config` is possible.

- `--system`: I have no idea.
- `--global`: possible with `$XDG_CONFIG_HOME/git/config`
- `--local`: possible with `$repo_url/.git/config` (current logic)
- `-f config-file`: possible with `GIT_CONFIG`.